File: ExtendedDataGridView Control/[VB]-ExtendedDataGridView Control/VB/ExtendedDataGridView/ExtendedDataGridView/DataGridViewColumns/altDataGridViewLinkColumn.vb
Imports System.ComponentModel
Imports System.Windows.Forms

Public Class altDataGridViewLinkColumn
    Inherits DataGridViewLinkColumn

    Private _HeaderStyle As enumerations.style1
    <Category("Design"), DesignerSerializationVisibility(DesignerSerializationVisibility.Visible)>
    Public Property HeaderStyle() As enumerations.style1
        Get
            Return _HeaderStyle
        End Get
        Set(ByVal value As enumerations.style1)
            _HeaderStyle = value
            Select Case _HeaderStyle
                Case style1.Standard
                    MyBase.HeaderCell = New DataGridViewColumnHeaderCell
                Case style1.HideColumn
                    MyBase.HeaderCell = New checkHideColumnHeaderCell
            End Select
        End Set
    End Property

    Public Overrides Function clone() As Object
        Dim copyColumn As altDataGridViewLinkColumn = DirectCast(MyBase.Clone, altDataGridViewLinkColumn)
        copyColumn._HeaderStyle = Me.HeaderStyle
        Return copyColumn
    End Function

End Class

File: ExtendedDataGridView Control/[C#]-ExtendedDataGridView Control/C#/ExtendedDatagridViewCS/DataGridViewColumns/altDataGridViewLinkColumn.cs
using System.Windows.Forms;
using System.ComponentModel;
namespace ExtendedDataGridView
{

    public class altDataGridViewLinkColumn : DataGridViewLinkColumn
	{

		private enumerations.style1 _HeaderStyle;
		[Category("Design"), DesignerSerializationVisibility(DesignerSerializationVisibility.Visible)]
		public enumerations.style1 HeaderStyle {
			get { return _HeaderStyle; }
			set {
				_HeaderStyle = value;
				switch (_HeaderStyle) {
					case enumerations.style1.Standard:
						base.HeaderCell = new DataGridViewColumnHeaderCell();
						break;
					case enumerations.style1.HideColumn:
						base.HeaderCell = new checkHideColumnHeaderCell();
						break;
				}
			}
		}

		public override object Clone()
		{
			altDataGridViewLinkColumn copyColumn = (altDataGridViewLinkColumn)base.Clone();
			copyColumn._HeaderStyle = this.HeaderStyle;
			return copyColumn;
		}

	}
}
